### PR TITLE
 WIP Refinedwhiff

### PR DESCRIPTION
**Pull request by mhhf**

### CODE REVIEW

File: src/hevm/hevm-cli/hevm-cli.hs [hunk 1]
```diff
@@ -729,7 +729,7 @@ vmFromCommand cmd = do
         value'   = word value 0
         caller'  = addr caller 0
         origin'  = addr origin 0
-        calldata' = ConcreteBuffer Oops $ bytes calldata ""
+        calldata' = ConcreteBuffer (Oops "cliCalldata") $ bytes calldata ""
```
Comment (rainbreak) [inline src/hevm/hevm-cli/hevm-cli.hs:732]: is it `clicalldata` or `cliCalldata`? looks inconsistent

File: src/hevm/src/EVM.hs [hunk 2]
```diff
@@ -786,7 +788,7 @@ exec1 = do
                 accessUnboundedMemoryRange fees memOffset n $ do
                   next
                   assign (state . stack) xs
-                  copyBytesToMemory (ConcreteBuffer Oops (the state code))
+                  copyBytesToMemory (ConcreteBuffer (Oops "codecopy") (the state code))
```
Comment (rainbreak) [inline src/hevm/src/EVM.hs:791]: upper case or lower case?

File: src/hevm/src/EVM.hs [hunk 3]
```diff
@@ -1880,7 +1882,7 @@ cheatActions =
   Map.fromList
     [ action "warp(uint256)" [AbiUIntType 256] $
         \_ _ [AbiUInt 256 x] ->
-          assign (block . timestamp) (sw256 $ num x),
+          assign (block . timestamp) (S (Dull "wrap") $ num x),
```
Comment (rainbreak) [inline src/hevm/src/EVM.hs:1885]: s/wrap/warp/

File: src/hevm/src/EVM/Format.hs [hunk 4]
```diff
@@ -399,34 +422,186 @@ showTree' (Node _ children) =
...
+
+propagateData :: (?srcInfo :: DappInfo) => Int -> Tree BranchInfo -> Tree BranchInfo
+propagateData _ n@(Node _ [])    = n
+propagateData _ (Node bi@(BranchInfo _ Nothing _) cs) = (Node bi (zipWith propagateData [0..] cs))
+propagateData 1 n@(Node bi@(BranchInfo vm (Just (IsZero (IsZero (Eq (Literal x) _)))) _) cs) =
+  let
+    abimap = view abiMap <$> currentSolc ?srcInfo vm
+    mmethod = (abimap >>= Map.lookup (num x))
+    mmm = mmethod
+  in case mmm of
```
Comment (MrChico) [inline src/hevm/src/EVM/Format.hs:604]:
```suggestion
  in case mmethod of
```

**Comment by d-xo** [comment]
Can this be closed? I think it's been superseded by https://github.com/dapphub/dapptools/pull/590 ?

**Comment by MrChico** [comment]
Sure

**Pull request closed without merging**